internal/services/mssqlmanagedinstance/mssql_managed_instance_resource.go

```diff
@@ -181,7 +180,7 @@ func (r MsSqlManagedInstanceResource) Arguments() map[string]*pluginsdk.Schema {
 			ValidateFunc: validate.ManagedInstanceID,
 		},
 
-		"identity": commonschema.SystemOrUserAssignedIdentityOptional(),
+		"identity": commonschema.SystemAssignedUserAssignedIdentityOptional(),
```

Comment:
This the crux of the identity bug fix, `SystemOrUserAssignedIdentityOptional` is XOR, exclusive or. `SystemAssignedUserAssignedIdentityOption` is true OR, i.e. one, the other, or both.